apikit-tools/apikit-scaffolder/src/main/java/org/mule/tools/apikit/Scaffolder.java

```diff
@@ -31,34 +38,40 @@
                                               List<String> specFiles, List<String> muleXmlFiles)
             throws IOException{
-        return createScaffolder(log, muleXmlOutputDirectory, specFiles, muleXmlFiles, null, null);
+        return createScaffolder(log, muleXmlOutputDirectory, specFiles, muleXmlFiles, null, null, null);
     }
 
-
     public static Scaffolder createScaffolder(Log log, File muleXmlOutputDirectory,
                                               List<String> specFiles, List<String> muleXmlFiles, String domainFile)
             throws IOException{
-        return createScaffolder(log, muleXmlOutputDirectory, specFiles, muleXmlFiles, domainFile, null);
+        return createScaffolder(log, muleXmlOutputDirectory, specFiles, muleXmlFiles, domainFile, null, null);
     }
 
     public static Scaffolder createScaffolder(Log log, File muleXmlOutputDirectory,
-                                              List<String> specFiles, List<String> muleXmlFiles, String domainFile, String muleVersion) throws IOException
+                                              List<String> specFiles, List<String> muleXmlFiles, String domainFile, String muleVersion)
+            throws IOException{
+        return createScaffolder(log, muleXmlOutputDirectory, specFiles, muleXmlFiles, domainFile, muleVersion, null);
+    }
+
+
+    public static Scaffolder createScaffolder(Log log, File muleXmlOutputDirectory, List<String> specPaths, List<String> muleXmlPaths, String domainPath, String muleVersion, List<String> ramlsWithExtensionEnabledPaths) throws IOException
     {
         FileListUtils fileUtils = new FileListUtils(log);
-        Map<File, InputStream> fileInputStreamMap = fileUtils.toStreamsOrFail(specFiles);
-        Map<File, InputStream> streams = fileUtils.toStreamsOrFail(muleXmlFiles);
-        InputStream domainStream = getDomainStream(log, domainFile);
-        return new Scaffolder(log, muleXmlOutputDirectory, fileInputStreamMap, streams, domainStream, muleVersion);
+        Map<File, InputStream> ramlStreams = fileUtils.toStreamsOrFail(specPaths);
+        Map<File, InputStream> muleStreams = fileUtils.toStreamsOrFail(muleXmlPaths);
+        Map<File, InputStream> ramlWithExtensionEnabled = fileUtils.toStreamsOrFail(ramlsWithExtensionEnabledPaths);
+        InputStream domainStream = getDomainStream(log, domainPath);
+        return new Scaffolder(log, muleXmlOutputDirectory, ramlStreams, muleStreams, domainStream, muleVersion, ramlWithExtensionEnabled);
     }
 
-    public Scaffolder(Log log, File muleXmlOutputDirectory,  Map<File, InputStream> ramls,
-                      Map<File, InputStream> xmls, InputStream domainStream, String muleVersion)  {
+    public Scaffolder(Log log, File muleXmlOutputDirectory,  Map<File, InputStream> ramls, Map<File, InputStream> xmls, InputStream domainStream, String muleVersion, Map<File, InputStream> ramlsWithExtensionEnabledPaths)
```

Comment:
rename `ramlsWithExtensionEnabledPaths` to `ramlFilesWithExtensionEnabled` and use a `Set<File>` instead of a `Map`